activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb

```diff
@@ -714,7 +714,29 @@ def reconnect
         def configure_connection
           @raw_connection.busy_timeout(self.class.type_cast_config_to_integer(@config[:timeout])) if @config[:timeout]
 
+          # Enforce foreign key constraints
+          # https://www.sqlite.org/pragma.html#pragma_foreign_keys
+          # https://www.sqlite.org/foreignkeys.html
           raw_execute("PRAGMA foreign_keys = ON", "SCHEMA")
+          unless @memory_database
+            # Journal mode WAL allows for greater concurrency (many readers + one writer)
+            # https://www.sqlite.org/pragma.html#pragma_journal_mode
+            raw_execute("PRAGMA journal_mode = WAL", "SCHEMA")
```

Comment:
For future readers: this feature is available in `3.7.0` and Active Record only supports `3.8.0+`: https://github.com/rails/rails/blob/e6f09150a88c1fdcfbf343226fe020837e665aab/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb#L423